### ISSUE TITLE
"should support rolling-update to same image" e2e test fails

### ISSUE THREAD

**Issue by oomichi** [kubernetes/kubernetes#59833]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

When operating conformance tests on my local environment, "should support rolling-update to same image" e2e test only failed always.
The following log was output many times and the timeout happened.

``
"Feb 13 20:02:41.088: INFO: e2e-test-nginx-rc-eda0488f68f5a4fdd5baf94cbc6dbf50-q7zbr is created but running wrong image; expected: k8s.gcr.io/nginx-slim-amd64:0.20, actual: gcr.io/google-containers/nginx-slim-amd64:0.20"
``

**What you expected to happen**:

Succeeded to operate all conformance tests.

**How to reproduce it (as minimally and precisely as possible)**:

Just run the following command under k8s.io/kubernetes:

``
$ go run hack/e2e.go -- --provider=skeleton -v --test --test_args="--ginkgo.focus=rolling-update"
``

**Anything else we need to know?**:

As the above log, the actual image is different from the expected like:
expected: k8s.gcr.io/nginx-slim-amd64:0.20
actual: gcr.io/google-containers/nginx-slim-amd64:0.20

According to the test scenario, this seems like that `kubectl get pods` shows gcr.io/google-containers/nginx-slim-amd64:0.20 as the image template, even if k8s.gcr.io/nginx-slim-amd64:0.20 was specified on `kubectl run`.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.9.2
- Cloud provider or hardware configuration: OpenStack
- OS (e.g. from /etc/os-release): ubuntu 16.04 LTS
- Kernel (e.g. `uname -a`): 4.4.0-112-generic
- Install tools: kubeadm
- Others:

**Comment by oomichi** [comment]
As the following, containerStatuses image is different from spec image on kubectl by hands.
Is that expected?

```
$ kubectl get pods e2e-test-nginx-rc-dj55c -o template --template='{{if (exists . "status" "containerStatuses")}}{{range .status.containerStatuses}}{{if eq .name "e2e-test-nginx-rc"}}{{.image}}{{end}}{{end}}{{end}}' --namespace=e2e-tests-kubectl-fjmlk -v=9
..
{"kind":"Pod","apiVersion":"v1",
 "metadata":{
   "name":"e2e-test-nginx-rc-dj55c","generateName":"e2e-test-nginx-rc-","namespace":"e2e-tests-kubectl-fjmlk",
   ..
 },
 "spec":{
   "containers":[
     {"name":"e2e-test-nginx-rc",
      "image":"k8s.gcr.io/nginx-slim-amd64:0.20",
   ..
 },
 "status":{
   "containerStatuses":[
     {"name":"e2e-test-nginx-rc",
      "state":{"running":{"startedAt":"2018-02-20T22:39:53Z"}},
      "image":"gcr.io/google-containers/nginx-slim-amd64:0.20",
      "imageID":"docker-pullable://gcr.io/google-containers/nginx-slim-amd64@sha256:6654db6d4028756062edac466454ee5c9cf9b20ef79e35a81e3c840031eb1e2b",
  ..
```

**Comment by oomichi** [comment]
At upstream CI, k8s.gcr.io 's image is used as the same as my local env.

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/59961/pull-kubernetes-e2e-gce/21446/build-log.txt
```
I0216 18:01:14.122] [It] should support rolling-update to same image  [Conformance]
I0216 18:01:14.122]   test/e2e/framework/framework.go:674
I0216 18:01:14.123] [1mSTEP [0m: running the image k8s.gcr.io/nginx-slim-amd64:0.20
```

So the containerStatuses' image seems wrong on my env. In addition, imageID also seems wrong like
```
valid: "ImageID":"docker-pullable://k8s.gcr.io/nginx-slim-amd64@.."
wrong: "imageID":"docker-pullable://gcr.io/google-containers/nginx-slim-amd64@..",
```

**Comment by oomichi** [comment]
This seems to be solved on k8s v1.10 env and the latest e2e.
